tests/unit/test_mysql_router_helpers.py

```diff
@@ -32,158 +28,119 @@
     "http_server.bind_address=127.0.0.1",
     "--conf-use-gr-notifications",
 ]
-chmod_cmd = [
+replace_socket_location_cmd = [
     "sudo",
-    "chmod",
-    "755",
-    "/var/lib/mysql/testapp",
+    "sed",
+    "-Ei",
+    "s:/tmp/(.+).sock:/var/snap/charmed-mysql/common/var/run/mysqlrouter/\\1.sock:g",
+    "/var/snap/charmed-mysql/current/etc/mysqlrouter/mysqlrouter.conf",
 ]
```

Comment:
General comment: having copy&paste duplicates in test and in charm.py for bootstrap_cmd and replace_socket_location_cmd leads to the human mistakes in the future. Can/should we extract it to the common resource?

Reply:
I agree, extracting it would defeat the point of having a test. I modified the existing tests so they pass, but the tests will need to be adjusted when carl's refactor is applied to this operator